pkg/kv/kvserver/replica_consistency.go

```diff
@@ -501,7 +501,7 @@ func CalcReplicaDigest(
 	var uuidBuf [uuid.Size]byte
 	hasher := sha512.New()
 
-	if efos, ok := snap.(storage.EventuallyFileOnlyReader); ok {
+	if efos, ok := snap.(storage.EventuallyFileOnlyReader); ok && !statsOnly {
```

Comment:
I think what you want to do is tune the efosWait down below to be shorter for `statsOnly`. The issue with not calling `WaitForFileOnly` at all is that you could see errors on iterator creation down the line if it conflicts with an excise (which now happens for all snapshot ingestions).

Reply:
Some naive questions, as I wasn't aware of this work and am just learning about it now:
- Will tuning this wait down have the effect of flushing the memtable almost immediately? If so, that doesn't seem great either.
- Where is the best place to read about this new form of snapshot?
- What happens if a conflicting excise hits between calls to `NewEventuallyFileOnlySnapshot` and `WaitForFileOnly`?
- What happens if a conflicting excise hits between during the call to `WaitForFileOnly`?
- How do excises interact with `EventuallyFileOnlySnapshot`s vs. with traditional `Snapshot`s?

Reply:
Those are all good questions - sorry for it not being clearer earlier. This comment should help describe how an EventuallyFileOnlySnapshot (EFOS) works: https://github.com/cockroachdb/pebble/blob/f0f5f40e8ab6e2d96aa0479e02e2cda0b356ebbd/snapshot.go#L215

* Yes, tuning the weight down will lead to excessive flushing - as long as there's something written in the range's key bounds that's in the memtable at the time of snapshot creation. Now I think we can reduce the frequency of this happening if we only specify the MVCC part of the range's bounds to EventuallyFileOnlySnapshot because that's the only part of a range that'll ever get excised anyway. That way, if only the local portion of a range has unflushed writes in memtables, an EFOS on it will be file-only from the start, turning WaitForFileOnly into a no-op. Maybe this needs to be an issue; I can take a stab at this.
* If there's a conflicting excise before WaitForFileOnly, WaitForFileOnly will return `pebble.ErrSnapshotExcised` and iterator creation will fail. The same is the case with excises during `WaitForFileOnly`, however if there's an excise after the transition to a file-only snapshot, the EFOS will remain immune as it has already grabbed its sstable references and can keep creating iterators no problem.
* EventuallyFileOnlySnapshot is an excise-aware Snapshot. A classic Snapshot can see keys disappear if a conflicting excise were to happen, and you created a new iterator on it after that excise. You can get around that caveat of a snapshot by holding onto an iterator, but then you lose all benefits of opening a Snapshot to begin with. The phasing-in of Excise instead of range deletes to clear out ranges is why we're forced to move to EFOS